lib/truffle/socket/ip_socket.rb

```diff
@@ -58,11 +58,13 @@ def peeraddr(reverse_lookup=nil)
       end
     end
 
+    buffer.replace(message) if buffer
```

Comment:
Similarly, here it should be:
```ruby
message = buffer.replace(message) if buffer
```
So that the first element returned is the buffer if given a buffer.

Reply:
done